js/components/visualizations/filter-panel/filter-panel.js

```diff
@@ -19,22 +19,25 @@ define(
       this.filterList = params.filterList;
       this.live = params.live || false;
       this.apply = params.apply;
-      this.clear = () => this.filterList.forEach(filter => filter.selectedValues([]));
+      this.clear = () => this.filterList().forEach(filter => filter.selectedValues([]));
 
       // Do not show "Apply" button in live mode and trigger filtering immediately
       if (this.live) {
-        Object.values(this.filterList).map(filter => {
-          let prevVal;
-          if (filter.type === 'select') {
-            prevVal = filter.selectedValue();
-            filter.selectedValue.subscribe(() => {
-              if (prevVal != filter.selectedValue()) {
-                prevVal = filter.selectedValue();
-                this.apply()
-              }
-            });
-          }
-        });
+        this.filterList.subscribe(newFilter => {
+          Object.values(newFilter).map(filter => {
+            let prevVal;
+            if (filter.type === 'select') {
+              prevVal = filter.selectedValue();
+              filter.selectedValue.subscribe(() => {
```

Comment:
subscribers should be disposed

Reply:
There should be a few examples in the codebase where you can see how a 'dispose' method is declared on the model, which will be triggered when the view is cleaned up via knockout.  As @johnSamilin says, subscriptions should be cleaned up, but also ko.computed (but not ko.pureComputed), etc.